decidim-conferences/app/commands/decidim/conferences/admin/copy_conference.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Conferences
+    module Admin
+      # A command with all the business logic when copying a new participatory
+      # conference in the system.
+      class CopyConference < Rectify::Command
+        # Public: Initializes the command.
+        #
+        # form - A form object with the params.
+        # conference - An conference we want to duplicate
+        def initialize(form, conference)
+          @form = form
+          @conference = conference
+        end
+
+        # Executes the command. Broadcasts these events:
+        #
+        # - :ok when everything is valid.
+        # - :invalid if the form wasn't valid and we couldn't proceed.
+        #
+        # Returns nothing.
+        def call
+          return broadcast(:invalid) if form.invalid?
+
+          Conference.transaction do
+            copy_conference
+            copy_conference_categories if @form.copy_categories?
+            copy_conference_components if @form.copy_components?
+          end
+
+          broadcast(:ok, @copied_conference)
+        end
+
+        private
+
+        attr_reader :form
+
+        def copy_conference
+          @copied_conference = Conference.create!(
```

Comment:
Why not use `conference.dup`? https://api.rubyonrails.org/classes/ActiveRecord/Core.html#method-i-dup

Reply:
We don't use almost anywhere '.dup'  in 'Decidim'. Maybe it's a decision that @decidim/lot-core  took at some point?

Reply:
@oriolgual the docs on the method specify that it does not transfer the timestamps nor the ID to the duplicated object.